app/src/main/res/layout/filter_dialog.xml

```diff
@@ -1,15 +1,35 @@
 <?xml version="1.0" encoding="utf-8"?>
-<ScrollView xmlns:android="http://schemas.android.com/apk/res/android"
-        android:layout_width="match_parent"
-        android:layout_height="match_parent">
+<ScrollView
+    xmlns:android="http://schemas.android.com/apk/res/android"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent">
+
     <LinearLayout
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:orientation="vertical">
+
+        <LinearLayout
             android:id="@+id/filter_rows"
             android:layout_width="match_parent"
             android:layout_height="wrap_content"
             android:orientation="vertical"
             android:paddingLeft="24dp"
             android:paddingTop="24dp"
             android:paddingRight="24dp"
-            android:paddingBottom="8dp">
+            android:paddingBottom="8dp" />
+
+        <TextView
```

Comment:
Please use a button with `style="@style/Widget.MaterialComponents.Button.TextButton"`. Then the ripple effect on touch works properly.